File: CMDB/CMDB.UI.Specflow/Questions/Identity/OpenTheCreateIdentityPage.cs
using Bright.ScreenPlay.Actors;
using CMDB.UI.Specflow.Abilities.Pages.Identity;
using Bright.ScreenPlay.Questions;
using CMDB.UI.Specflow.Abilities.Pages;

namespace CMDB.UI.Specflow.Questions.Identity
{
    public class OpenTheCreateIdentityPage : Question<CreateIdentityPage>
    {
        public override CreateIdentityPage PerformAs(IPerformer actor)
        {
            var page = actor.GetAbility<IdentityOverviewPage>();
            page.ClickElementByXpath(MainPage.NewXpath);
            CreateIdentityPage createIdentityPage = WebPageFactory.Create<CreateIdentityPage>(page.WebDriver);
            return createIdentityPage;
        }
    }
}

File: Java/playwright/src/test/java/be/hans/cmdb/Questions/OpenTheCreateIdentityPage.java
package be.hans.cmdb.Questions;

import be.brightest.ScreenPlay.Abilities.WebPageFactory;
import be.brightest.ScreenPlay.Actors.IActor;
import be.brightest.ScreenPlay.Question.Question;
import be.cmdb.pages.Identity.CreateIdentityPage;
import be.cmdb.pages.Identity.IdentityOverviewPage;

public class OpenTheCreateIdentityPage extends Question<CreateIdentityPage> {
    @Override
    public CreateIdentityPage performAs(IActor actor) {
        IdentityOverviewPage ability = (IdentityOverviewPage) actor.getAbility(IdentityOverviewPage.class);
        ability.newButton().click();
        ability.getPage().waitForLoadState();
        return WebPageFactory.create(CreateIdentityPage.class, ability.getPage());
    }
}
